### DISCUSSION TITLE
X/Y resolution question

### DISCUSSION THREAD

**Discussion by waylander131** [arkypita/LaserGRBL#2078] [General]
Hello, community,

I have recently gotten into the laser engraving world, I have built my CNC myself, using two Y-steppers, one X, manual Z control, and Laser-Tree 80W LED laser (10W optical). I've chosen LaserGRBL as the best available software I can try for free and I've set up my machine to a resolution of 40 steps per mm in X and Y directions, and it runs fine. 
I am engraving some backlit pushbuttons, but because my text is small, I can see the effect of the coarse resolution of LaserGRBL software. I will try to attach a few screenshots to illustrate my problem. Basically, despite having the machine set up for 40 steps per mm, in reality, I got an effective resolution of 4 steps per mm.
What I do is create my object in Inkscape, then export it to a PNG image and import it to LaserGRBL. It is possible that's not the correct procedure, but that gives the best results so far.
Is there any way I can increase the LaserGRBL resolution or any other solution?
I will be very grateful if someone can provide me with a tip.
Regards,
Stefan
![resolution XY](https://github.com/arkypita/LaserGRBL/assets/30437872/611eb662-43dd-4dd9-afa9-c3178b346a90)
![Original Inkscape image](https://github.com/arkypita/LaserGRBL/assets/30437872/250ed863-8929-4ba9-9474-5c16e50694d4)

**Comment by Advocas** [comment]
The best way I found to engrave text is as follows

- You will need to install the Eggbot third party extension to fill the text it can be found here [https://wiki.evilmadscientist.com/The_Eggbot_Extensions](url)
- Open Inkscape
- Create the text
- If it's filled remove the fill (LaserGRBL ignores fill)
- Convert it to a Path (Path, Object to Path)
- Run the Hatch fill Eggbot extention
- ![image](https://github.com/arkypita/LaserGRBL/assets/89773859/c212e725-247e-4aec-bd39-aa4f1f94d509)
- ![image](https://github.com/arkypita/LaserGRBL/assets/89773859/b3fef1f8-86d1-43e0-a6e4-f8b84f1546ec)
- Save as a Vector (.svg)
- Load into LaserGRBL and engrave

**Reply by waylander131** [reply]
Many thanks Advocas, I will try your solution later today, it looks promising :)

**Comment by Tribo-2dot0** [comment]
Em primeiro lugar eu configuraria os motores de passo para 1/16 passos e usaria 80 passos/mm nas configurações grbl.
Em Inkscape ao exportar o texto como png, exporte com uma boa resolução, algo como 318dpi.
Em LaserGrbl, nas opções de Vectorize, você tem a opção de marcar "Adaptive quality".
Uma boa opção é o que Advocas sugeriu, usando Hatch Fill e salvando como svg.

**Comment by arkypita** [comment]
With Hi-Res filling LaserGRBL is able to fill up to 50 lines/mm but if you don't match request size with output size you may need to correct your GRBL step/mm configuration.

![image](https://github.com/arkypita/LaserGRBL/assets/8782035/30fe7d2e-fb28-47ef-82a7-c7301948af32)

**Comment by waylander131** [comment]
Thanks again everyone for your tips. I enabled the Hi-Res Filling and also installed the eggbot extension. Had to spend some time finding the correct font for my project, the best match turned out to be Futura MdCN BT, which after the Object-to-Path command left clean edges, and the eggbot did excellent job in filling those letters with hatches. Also now I have no issues opening the SVG file in LaserGRBL!
Great stuff, I now love my little laser CNC :)
